WordPress/Classes/ViewRelated/Blog/SharingAccountViewController.swift

```diff
@@ -0,0 +1,292 @@
+        }
+
+        // Build the section for unconnected accounts
+        var rows = rowsForUnconnectedKeyringAccounts(accounts)
+        if let section = sectionForUnconnectedKeyringAccountRows(rows) {
+            sections.append(section)
+        }
+
+        // Build the section for connected accounts
+        rows = rowsForConnectedKeyringAccounts(connectedAccounts)
+        if rows.count > 0 {
+            let title = NSLocalizedString("Connected", comment: "Adjective. The title of a list of third-part sharing service account names.")
+            let section = ImmuTableSection(headerText: title, rows: rows, footerText: nil)
+            sections.append(section)
+        }
+
+        return ImmuTable(sections: sections)
+    }
+
+
+    /// Builds the ImmuTableSection that displays unconnected keyring accounts. 
+    ///
+    /// - Parameters: 
+    ///     - rows: An array of ImmuTableRow objects appearing in the section.
+    ///
+    /// - Returns: An ImmuTableSection or `nil` if there were no rows.
+    ///
+    private func sectionForUnconnectedKeyringAccountRows(rows: [ImmuTableRow]) -> ImmuTableSection? {
+        if rows.count == 0 {
+            return nil
+        }
+
+        var title =  NSLocalizedString("Connecting %@", comment: "Connecting is a verb. Title of Publicize account selection. The %@ is a placeholder for the service's name");
+        title = NSString(format: title, publicizeService.serviceID) as String
+
+        let manyAccountFooter = NSLocalizedString("Select the account you would like to authorize. Note that your posts will be automatically shared to the selected account.", comment: "")
+        let oneAccountFooter = NSLocalizedString("Confirm this is the account you would like to authorize. Note that your posts will be automatically shared to this account.", comment: "")
+        let footer = rows.count > 1 ? manyAccountFooter : oneAccountFooter
+
+        return ImmuTableSection(headerText: title, rows: rows, footerText: footer)
+    }
+
+
+    /// Builds the ImmuTableSection that displays connected keyring accounts.
+    ///
+    /// - Parameters:
+    ///     - rows: An array of ImmuTableRow objects appearing in the section.
+    ///
+    /// - Returns: An ImmuTableSection or `nil` if there were no rows.
+    ///
+    private func rowsForUnconnectedKeyringAccounts(accounts: [KeyringAccount]) -> [ImmuTableRow] {
+        var rows = [ImmuTableRow]()
+        for acct in accounts {
+            let row = KeyringRow(title: acct.name, value: "", action: actionForRow(acct));
+
+            rows.append(row)
+        }
+
+        return rows
+    }
+
+
+    /// Builds an ImmuTableAction that should be performed when a specific row is selected.
+    ///
+    /// - Parameters:
+    ///     - The keyring account for the row.
+    ///
+    /// - Returns: An ImmuTableAction instance.
+    ///
+    private func actionForRow(keyringAccount: KeyringAccount) -> ImmuTableAction {
+        return { [unowned self] row in
```

Comment:
Gotcha!. Yeah, in this particular case i think it's allright. Asking, out of curiosity... because... unowed opens the door for invalid reference crashes (Mwah ha ha!)

=D